test/controllers/sessions_controller_test.rb

```diff
@@ -46,15 +46,32 @@ class SessionsControllerTest < ActionController::TestCase
     assert_equal @user.id, current_user.id
   end
 
-  test 'should not create a new session' do
+  test 'should not create a new session with correct user_name' do
     Ldap.default.destroy!
 
     assert_no_difference '@user.logins.count' do
-      post :create, params: { username: @user.email, password: 'wrong' }
+      assert_difference 'Fail.all.count' do
```

Comment:
```suggestion
      assert_difference '@user.fails.count' do
```
Esto te va a "obligar" a agregar el `has_many` en el user, pero para asegurarnos que lo asocia y demás.

Reply:
okey, me parece bien, asi tenemos la relacion bidireccional. Le agrego el has_many entonces.